### PR TITLE
move storage/api to its own module

### PR DESCRIPTION
**Pull request by rolfyone**
move most of the storage/api to its own module.

 - the StorageUpdateChannel is a bit more sticky, and haven't moved that. The Eth1DepositStorageChannel is the main requirement for the rest of 1412.

### CODE REVIEW

File: sync/build.gradle [hunk 1]
```diff
@@ -9,6 +9,7 @@ dependencies {
     implementation project(':services:serviceutils')
     implementation project(':ssz')
     implementation project(':storage')
```
Comment (ajsutton) [inline sync/build.gradle:11]: nit: I wonder if we can remove a bunch of these dependencies on storage now and only depend on storage:api?

**Review by ajsutton — APPROVED** [review]
LGTM.

File: sync/build.gradle (hunk 1 above)
Comment (rolfyone) [reply sync/build.gradle:11]: probably if they're not using storage/server... i'll take a quick look before i merge

Comment (rolfyone) [reply sync/build.gradle:11]: we'll need to break out storage:server and storage:client, then we can remove a lot of the dependencies, and also the remaining api member in the main server folder could be moved after that.

**Pull request merged**